asdf/_core/_converters/ndarray.py

```diff
@@ -64,8 +54,22 @@ def to_yaml_tree(self, obj, tag, ctx):
         else:
             options = ctx._blocks.options.get_options(data)
 
-        # possibly override options based on config settings
         cfg = config.get_config()
+
+        # The view computations that follow assume that the base array
+        # is contiguous.  If not, we need to make a copy to avoid
+        # writing a nonsense view.
+        if options.save_base or (options.save_base is None and cfg.default_array_save_base):
+            base = util.get_array_base(data)
+        else:
+            base = data
+        if not base.flags.forc:
+            data = np.ascontiguousarray(data)
+            base = util.get_array_base(data)
+
+        shape = data.shape
```

Comment:
Yeah, I think this fixes a niche problem (and another one could easily be fixed, move on that below). If with the current main branch we try to save a non-"forc" array it will ignore storage options
```python
import asdf, numpy as np
arr = np.zeros((4, 4))
arr.stries = (4, 4)  # make this non-"forc" without making it a view
af = asdf.AsdfFile({'arr': arr})
af.set_array_storage(arr, 'inline')
af.write_to('foo.asdf')  # foo.asdf will put arr in an internal (not inline) block
```
With this PR the options will be respected.

If we similarly move the check for 0 strides we could fix the issue for those arrays as well. I'll update this PR.